### PR TITLE
has_ships function optimized

### PR DESCRIPTION
**Pull request by GeRmAnImAl**
I'm new to python, VSC, and GitHub but I think I was able to optimize the has_ships function. Disregard the changes made in ship.py, when playing with the code I ended up adding an import from Player. This unused import was removed.

### CODE REVIEW

File: player.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ def __init__(self, name, shipPlacementStrategy: IShipPlacementStrategy) -> None:
         self.name = name
         self._score = 0
         self._hits = 0
+        self.num_ships = 4
```
Comment (Baso1st) [inline player.py:18]: Since the game is designed to have any number of ships. It would be a good idea if we set num_ships through the Player constructor.

**Comment by Baso1st** [comment]
@GeRmAnImAl  Thank you very much for the PR, It will work for sure. I added a couple of comments for further improvements, but I will go ahead and approve your code. 
Thank you again :)

**Comment by GeRmAnImAl** [comment]
No problem, I saw your comments and I can adjust the code to account for that. Are you sure you don't want an upper limit to the amount of ships they can have? There are only so many spaces on a board 😅

**Comment by Baso1st** [comment]
I see your point about the upper limit 😊, but it is something that could be handled on the clients side if we are thinking from an API perspective or in our case of a console game we can just take care of it in the gameloop class.

**Pull request merged**